### ISSUE TITLE
I2C devices fail to initialize with clock-frequency = I2C_BITRATE_FAST_PLUS on nrf5340

### ISSUE THREAD

**Issue by Yaxit** [zephyrproject-rtos/zephyr#67724]
_Disclaimer: I would have initially filed this issue in the Nordic repository however it seems that issues have been disabled on their end. Correct me if I'm wrong._

**Describe the bug**
Setting the clock-frequency property to I2C_BITRATE_FAST_PLUS causes the I2C device to not initialize on a nrf5340 platform. I encountered this issue when upgrading my firmware from NCS v2.3.0 to v2.5.1, and I could replicate it with a custom board and with the nrf5340DK board.
This speed should however be supported.

**To Reproduce**
With the DK board
app.overlay file
```dts
&i2c1 {
	status = "okay";
	pinctrl-0 = <&i2c1_default>;
	pinctrl-1 = <&i2c1_sleep>;
	pinctrl-names = "default", "sleep";
        zephyr,concat-buf-size = <32>;
        # works with I2C_BITRATE_FAST for example
	clock-frequency = < I2C_BITRATE_FAST_PLUS >;
};
```
prj.conf file
```
CONFIG_I2C=y
CONFIG_LOG=y
CONFIG_DEBUG=y

CONFIG_LOG=y
CONFIG_CONSOLE=y
CONFIG_UART_CONSOLE=n
CONFIG_RTT_CONSOLE=y
CONFIG_USE_SEGGER_RTT=y
CONFIG_LOG_BACKEND_RTT=y
CONFIG_SEGGER_RTT_BUFFER_SIZE_UP=8192
CONFIG_SEGGER_RTT_BUFFER_SIZE_DOWN=8192

CONFIG_ASSERT=y
CONFIG_ASSERT_VERBOSE=y
```
an empty main file is enough.

**Expected behavior**
The speed of 1Mbps should be supported

**Impact**
FAST_PLUS speed is not supported, updates might silently break firmware if assertions are not enabled.

**Logs and console output**
```
[00:00:00.489,257] <err> i2c_nrfx_twim: Failed to initialize device: i2c@9000
*** Booting Zephyr OS build zephyr-v3.5.0-4318-g9569e4410475 ***
ASSERTION FAIL [p_cb->state == NRFX_DRV_STATE_POWERED_ON] @ WEST_TOPDIR/modules/hal/nordic/nrfx/drivers/src/nrfx_twim.c:618
[00:00:00.489,379] <err> os: r0/a1:  0x00000004  r1/a2:  0x0000026a  r2/a3:  0x00000002
[00:00:00.489,379] <err> os: r3/a4:  0x0000026a r12/ip:  0x00000004 r14/lr:  0x00007ddb
[00:00:00.489,410] <err> os:  xpsr:  0x69000000
[00:00:00.489,410] <err> os: Faulting instruction address (r15/pc): 0x0000c2ee
[00:00:00.489,440] <err> os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
[00:00:00.489,471] <err> os: Current thread: 0x200006d8 (unknown)
[00:00:01.562,835] <err> os: Halting system
```

**Environment (please complete the following information):**
 - Windows
 - Zephyr SDK 0.16
 - Zephyr main 9569e4410475fb62125e769a13222882902d8c75

**Comment by henrikbrixandersen** [comment]
Thank you for reporting this. However - unless you are able to reproduce this issue with upstream Zephyr `main` - please report issues with the nRF Connect SDK (NCS) on the [Nordic Semiconductor DevZone](https://devzone.nordicsemi.com/).

Also, please use our bug template when reporting bugs. If you can reproduce this with upstream Zephyr, please edit this issue to include the information requested in https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/ISSUE_TEMPLATE/001_bug_report.md

**Comment by Yaxit** [comment]
I could replicate this on main Zephyr (9569e44104) with a nrf5340DK board, however it seems I initially did not correctly identified the problem, as I forgot to do a pristine build after changing the property in the DT.
The issue seems to be related to the speed rather than the concatenation buffer. When `clock-frequency` is set to `I2C_SPEED_FAST_PLUS` (1Mbps) which should be supported, the device fails to initialize.
Should I open a new issue or adapt this one?

**Comment by henrikbrixandersen** [comment]
> Should I open a new issue or adapt this one?

Feel free to adapt this one.

**Comment by anangl** [comment]
Fixed by #72586.
